src/storage-client/src/controller/metrics_scraper.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A tokio task (and support machinery) for maintaining storage-managed
+//! collections.
+
+use std::any::Any;
+use std::sync::Arc;
+use std::time::Duration;
+
+use itertools::Itertools;
+use timely::progress::ChangeBatch;
+use tokio::sync::oneshot;
+
+use mz_ore::cast::CastFrom;
+use mz_repr::{Datum, GlobalId, Row};
+
+use crate::controller::collection_mgmt::CollectionManager;
+use crate::controller::hosts::MetricsFetcher;
+
+/// Spawns a task that continually (at an interval) fetches metrics from the
+/// given [`MetricsFetcher`] and writes it out to the collection identified by
+/// `metrics_collection_id`. The task will be stopped when the returned token is
+/// dropped.
+///
+/// This assumes that the output collection is empty (or at least that updates
+/// sum up to a zero diff) and that we are the sole writer. Once we need support
+/// for concurrency we need to either teach [`CollectionManager`] to deal with
+/// that or we need to learn to reconciliate our knowledge about the contents of
+/// the output shard with other processes, by reading back in the current
+/// contents of the output shard.
+pub(super) fn spawn_metrics_scraper(
+    metrics_collection_id: GlobalId,
+    mut metrics_fetcher: MetricsFetcher,
+    collection_mgmt: CollectionManager,
+) -> Arc<dyn Any + Send + Sync> {
+    // TODO(aljoscha): Should this be configurable? Maybe via LaunchDarkly?
+    const METRICS_INTERVAL: Duration = Duration::from_secs(10);
```

Comment:
module level?

Reply:
I don't know, compute also does it like this, so I'm inclined to just punt and leave as is. 🤷